### DISCUSSION TITLE
Route not found when in production

### DISCUSSION THREAD

**Discussion by jvbraganca** [supabase/supabase#368] [Questions] [answered]
Hello,

I am using supabase-js for the API connection. On my local machine using docker or node server I am able to retrieve all the data correctly but when I try on my production server I get this JSON:

```js
{
    "error": {
        "message": "no Route matched with those values"
    },
    "data": null,
    "status": 404,
    "statusText": "Not Found",
    "body": null
}
```
Is this a known problem? The API is online and the requests are being made on the correct API. Can someone give me some light, please?

I am using Express for powering the API. Here is the `package.json`

```js
"dependencies": {
    "@apidevtools/swagger-parser": "^10.0.2",
    "@sendgrid/mail": "^6.5.4",
    "@supabase/supabase-js": "^1.0.7",
    "body-parser": "^1.19.0",
    "consign": "^0.1.6",
    "cors": "^2.8.5",
    "dotenv": "^8.1.0",
    "express": "^4.17.1",
    "mongoose": "^5.7.1",
    "morgan": "^1.10.0",
    "request": "^2.88.0",
    "socket.io": "^2.2.0",
    "swagger-jsdoc": "^6.0.0-rc.3",
    "swagger-parser": "^10.0.2",
    "swagger-ui-express": "^4.1.5",
    "uuid": "^8.3.2"
  }
```

**Comment by steve-chavez** [comment]
Hey @jvbraganca,

>  "message": "no Route matched with those values"

What `supabase-js` function are you calling for getting that error?

**Reply by jvbraganca** [reply]
Hello @steve-chavez,

I am calling this one: 

```js
const body = await supabase
      .from('chats')
      .select('*')
      .or(`from.eq.${id},to.eq.${id}`)
```

**Reply by steve-chavez** [reply]
Ok. And how are you initializing the supabase client?

Can you double check you put the right URL here:

```js
import { createClient } from '@supabase/supabase-js'

const supabase = createClient('URL', 'public-anon-key')
```

The `URL` should be the one that shows in your `/settings/api` page.

**Reply by jvbraganca** [reply]
On ./lib/Store.js :

```js
const { createClient } = require('@supabase/supabase-js')

exports.supabase = createClient(
  process.env.SUPABASE_CHAT_URL,
  process.env.SUPABASE_CHAT_KEY
)
```

And importing it on the controller as:
```js
const { supabase } = require('../../lib/Store')
```

The supabase on my local machine is working but when I try it on the production I get the route not found thing. Here is the answer I get making the same request on my local machine:
```js
{

    "error": null,
    "data": [
        {
            "id": "bc5cc363-5d06-4820-8350-5f8dba146e11",
            "title": "Hello, World!",
            "from": "5efaaaa76dd03a016c68236e",
            "to": "21232f297a57a5a743894a0e4a801fc3",
            "updated_at": "2020-12-09T15:57:19.50431+00:00",
            "created_at": "2020-12-09T15:57:19.50431+00:00",
            "country": "BR"
        }
    ],
    "status": 200,
    "statusText": "OK",
    "body": [
        {
            "id": "bc5cc363-5d06-4820-8350-5f8dba146e11",
            "title": "Hello, World!",
            "from": "5efaaaa76dd03a016c68236e",
            "to": "21232f297a57a5a743894a0e4a801fc3",
            "updated_at": "2020-12-09T15:57:19.50431+00:00",
            "created_at": "2020-12-09T15:57:19.50431+00:00",
            "country": "BR"
        }
    ]
}
```

**Reply by kiwicopple** [reply]
I suspect that this isn't working in production:

```
exports.supabase = createClient(
  process.env.SUPABASE_CHAT_URL,
  process.env.SUPABASE_CHAT_KEY
)
```

What framework are you using? Often the builder with require some prefixes like `REACT_APP_SUPABASE_CHAT_URL`. 

Can you verify the URL and Key here, and not getting a null value?

**Reply by jvbraganca** [reply]
Hello @kiwicopple, had hoped it was this but I logged the variables and got the correct values. 

I am only using express for this one, I am building an API.

**Reply by steve-chavez** [reply]
@jvbraganca Could you try making the request with curl to see if you got the right URL?

There are some sample `curl` calls on your api page `/api/default`(click the bash tab)

**Comment by jvbraganca** [comment, marked as answer]
@steve-chavez with curl I was able to retrieve the data. Any idea why I am not with express?

The request:
```js
curl 'https://tbqtyripmdyrulzpwzvb.supabase.co/rest/v1/chats?select=*' \
-H "apikey: SUPABASE_KEY" \
-H "Authorization: Bearer SUPABASE_KEY"
The response:
```
```js
[
   {
      "id": "bc5cc363-5d06-4820-8350-5f8dba146e11",
      "title": "Hello, World!",
      "from": "5efaaaa76dd03a016c68236e",
      "to": "21232f297a57a5a743894a0e4a801fc3",
      "updated_at": "2020-12-09T15:57:19.50431+00:00",
      "created_at": "2020-12-09T15:57:19.50431+00:00",
      "country": "BR"
   }
]
```

UPDATE: Wait I copied the URL from the Supabase page.

**UPDATE 2: Checked again my URL and I was getting this error because of an extra `/`.**

Feeling blind as hell! Thank you, guys!
